app/models/person.rb

```diff
@@ -113,6 +113,11 @@ class Person < ActiveRecord::Base
 
   has_and_belongs_to_many :followed_listings, :class_name => "Listing", :join_table => "listing_followers"
 
+  after_save do
+    # This is required to invalidate the fragment caches for listings on front page, since the author is shown there.
+    self.listings.all.map(&:touch)
+  end
+
```

Comment:
This is the problematic part, but shouldn't cause any issues unless people with a lot of listings update their profile a lot. With `belongs_to :community` this could be skipped, and the update chain would be `listing` -> `person` -> `community`.